classes/PaymentModule.php

```diff
@@ -406,7 +406,7 @@ public function validateOrder($id_cart, $id_order_state, $amount_paid, $payment_
                     $transaction_id = null;
                 }
 
-                if (!$order->addOrderPayment($amount_paid, null, $transaction_id)) {
+                if (!isset($order) || !$order->addOrderPayment($amount_paid, null, $transaction_id)) {
```

Comment:
I will add this check to, but need to kkep the isset in reference to this bug http://forge.prestashop.com/browse/PSCSX-5576